jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/container/ContainerConfigurationHandler.java

```diff
@@ -0,0 +1,42 @@
+package io.opentelemetry.contrib.jfr.metrics.container;
+
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.contrib.jfr.Constants;
+import io.opentelemetry.contrib.jfr.metrics.RecordedEventHandler;
+import jdk.jfr.consumer.RecordedEvent;
+
+public class ContainerConfigurationHandler implements RecordedEventHandler {
+    public static final String EVENT_NAME = "jdk.ContainerConfiguration";
+    public static final String JFR_CONTAINER_CONFIGURATION = "jfr.ContainerConfiguration";
+
+    private static final String EFFECTIVE_CPU_COUNT = "effectiveCpuCount";
+
+    private final Meter otelMeter;
+    private volatile long value = 0L;
+
+
+    public ContainerConfigurationHandler(Meter otelMeter) {
+        this.otelMeter = otelMeter;
+    }
+
+    public ContainerConfigurationHandler init() {
+        otelMeter.upDownCounterBuilder(JFR_CONTAINER_CONFIGURATION)
+                .ofDoubles()
+                .setUnit(Constants.PERCENTAGE)
+                .buildWithCallback(codm -> codm.observe(value));
+
+        return this;
+    }
+
+        @Override
+    public String getEventName() {
+        return EVENT_NAME;
+    }
+
+    @Override
+    public void accept(RecordedEvent ev) {
+        if (ev.hasField(EFFECTIVE_CPU_COUNT)) {
+            value = ev.getLong(EFFECTIVE_CPU_COUNT);
```

Comment:
Just want to confirm, are these metrics not present in management beans? Is there anything conceptually different between metrics on jfr events vs management beans?

I ask mostly because if we end up memoizing to a field anyways, then there wouldn't be a real difference between using jfr metrics vs reading from a bean I guess, assuming the metric is actually available in a bean

Reply:
JFR events are not exactly the same as JMX values. They are basically better quality data, but I think a broader discussion of how to handle the multiplicity of data sources is required. Let's have that wider discussion after this initial agent is landed.